src/solvers/kriging.jl

```diff
@@ -98,34 +99,30 @@ function preprocess(problem::EstimationProblem, solver::Kriging)
 
       # determine path and neighborhood search method
       if varparams.maxneighbors ≠ nothing
-        # locations with data for variable
-        varlocs = collect(keys(datamap(problem, var)))
-
         if varparams.neighborhood ≠ nothing
           # local search with a neighborhood
           neigh = varparams.neighborhood
 
-          # create a path from the data and outwards
-          # use at most 10^2 points to generate path
-          N = length(varlocs); M = ceil(Int, N/10^2)
-          path = SourcePath(view(varlocs,1:M:N))
-
-          searcher  = NeighborhoodSearcher(pdomain, neigh)
-          bsearcher = BoundedSearcher(searcher, maxneighbors)
+          if neigh isa BallNeighborhood
+            #for specialised knn and ball search combined
+            bsearcher = KBallSearcher(pdata, maxneighbors,neigh)
+          else
+            searcher  = NeighborhoodSearcher(pdata, neigh)
+            bsearcher = BoundedSearcher(searcher, maxneighbors)
+          end
         else
           # nearest neighbor search with a distance
           distance = varparams.distance
           path = LinearPath()
-          bsearcher = NearestNeighborSearcher(pdomain, maxneighbors,
-                                              locations=varlocs, metric=distance)
+          bsearcher = NearestNeighborSearcher(pdata, maxneighbors,metric=distance)
```

Comment:
Extra space after last comma is missing.

Reply:
There are many comma missing around the code. Should I take care of the code that I just wrote?

Reply:
Please feel free to fix any missing comma you find in the source.